oauthenticator/gitlab.py

```diff
@@ -88,27 +80,69 @@ def _authorize_url_default(self):
     def _token_url_default(self):
         return f"{self.gitlab_url}/oauth/token"
 
+    gitlab_api_version = CUnicode(
```

Comment:
Small typo:

```suggestion
    gitlab_api_version = Unicode(
```

Reply:
It isn't a change in this PR to add the `C`, but I'm not sure about the type. I think its related to converting from one type to another.

Looking at https://traitlets.readthedocs.io/en/stable/trait_types.html#traitlets.CUnicode, its not so clear:

> Casting variants. When a value is assigned to the attribute, these will attempt to convert it to their type. They will not automatically encode/decode between unicode and bytes, however.

What does "convert it to their type" really mean? I figure it becomes an int if the value assigned is a string `"4"`?

Reply:
Figured out in https://github.com/jupyterhub/oauthenticator/pull/627#discussion_r1246840709, this is as it should be!